### PR TITLE
Tran 148 implement password change service

### PR DESCRIPTION
**Pull request by amysiv42**
This pull request introduces a new `changePassword` feature to the authentication service and refactors related data access and service interfaces to support it. It also updates tests to cover the new functionality and ensures consistent method naming for user lookup. The most important changes are summarized below:

**New Feature: Change Password**

* Added a `changePassword` method to `AuthService`, allowing users to securely change their password after verifying their current credentials. This includes error handling for user not found, credentials not found, incorrect current password, and updates to refresh tokens.

* Introduced the `ChangePasswordDto` type for the new password change operation.
* Added relevant error messages for missing user credentials. 

**Refactoring and Interface Updates**

* Changed the `findByUserId` method in `CredentialsDao` and its interface to return the full `UserCredentials` object instead of just the hashed password, enabling more flexible credential handling. 
* Renamed `findUserById` to `findByUserId` in the `UserManagementService` interface and all usages for consistency.

**Testing**

* Added a comprehensive test suite for the `changePassword` method, covering success, error cases, and edge conditions. 
* Updated existing tests for login and credential DAO to match new return types and method signatures.

**Type and Import Updates**

* Updated imports to include new DTOs and types required for password change and credential handling.
* Improved comments for DTOs to clarify their purpose.

### CODE REVIEW

File: src/auth/src/services/auth.ts [hunk 1]
```diff
@@ -37,34 +39,109 @@ export class AuthService {
...
+    if (!oldCredentials) {
+      throw new ResourceNotFoundError(AUTH_ERROR_MESSAGES.USER_CREDENTIAL_NOT_FOUND_BY_ID(data.userId));
+    }
+    if (!(await comparePasswordHash(data.currentPassword, oldCredentials.hashedPassword))) {
+      throw new AuthenticationError(AUTH_ERROR_MESSAGES.INVALID_CREDENTIALS);
+    }
+    const newHashedPassword = await passwordHasher(data.newPassword, SaltLimits);
+    await this.credentialsDao.updatePassword({ userId: data.userId, newPassword: newHashedPassword });
+    await this.refreshTokenDao.purgeRevokedExpired();
+  }
```
[the remark this answers is not in the document]
Comment (amysiv42) [reply src/auth/src/services/auth.ts]: ok
